contracts/core/base/Locked721Base.sol

```diff
@@ -0,0 +1,70 @@
+// SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.2;
+
+import "./Locked721AccessControl.sol";
+
+abstract contract Locked721Base is Locked721AccessControl {
+    // stores the locked state for each NFT
+    mapping(uint256 => bool) tokenLock;
+
+    constructor() {
+      // contract deployer is the admin by default
+      _setupRole(DEFAULT_ADMIN_ROLE, msg.sender);
+      // contract deployer can be granted some initial roles to start with
+      _setupRole(API_DELEGATE_ROLE, msg.sender);
+
+      // set default admin as the manager of claim and unlock delegate roles
+      _setRoleAdmin(API_DELEGATE_ROLE, DEFAULT_ADMIN_ROLE);
+    }
+
+    // TOKENLOCK FUNCTIONS
+    function _setTokenLock(uint256 tokenId, bool locked) internal {
+      tokenLock[tokenId] = locked;
+    }
+
+    function setTokenLock(uint256 tokenId, bool locked) public onlyApiDelegate {
+      _setTokenLock(tokenId, locked);
+    }
+
+    // READ TOKENLOCK STATE FUNCTIONS
+    function _getTokenLock(uint256 tokenId) internal virtual view returns (bool) {
+      return tokenLock[tokenId];
+    }
+
+    function getTokenLock(uint256 tokenId) public view returns (bool) {
+      return _getTokenLock(tokenId);
+    }
+
+    // if a token is locked, then this function returns true, 
+    // unless the token is being minted,
+    // or the message sender has an API DELEGATE role and is claiming the token,
+    // or the token is being burned 
+    function _shouldPreventTokenTransfer (address from, address to, uint256 startTokenId, uint256 batchSize) internal virtual view returns (bool) {
+      // do not prevent token transfer if:
+      if (from == address(0) || // token is being minted
+        (from == msg.sender && hasRole(API_DELEGATE_ROLE, msg.sender)) || // transfer is being initiated by API_DELEGATE_ROLE
+        to == address(0) // token is being burned
+      ) {
+        return false;
+      }
+      for (uint256 i=0; i<batchSize; i++) {
+        if (_getTokenLock(startTokenId + i)) {
+          return true;
+        }
+      }
+      return false;
+    }
+
+    function _lockTokensAfterTransfer (address from, address, uint256 startTokenId, uint256 quantity) internal {
+      if (from != address(0)) { // if token is being minted
```

Comment:
So now, anytime a token is minted, it will always be unlocked. Is there ever an edge case where we would want to mint into a locked state?

Reply:
i've been minting them into the locked state actually, so the actual NFT can be revealed after they claim and unlock and its not publicly on opensea.